pysd/py_backend/model.py

```diff
@@ -1690,33 +1710,329 @@ def _integrate(self, capture_elements):
+        present in the capture elements to it.
+
+        Parameters
+        ----------
+        model: pysd.Model
+            PySD Model object
+        capture_elements: set
+            Which model elements to capture - uses pysafe names.
+
+        Returns
+        -------
+        None
+
+        """
+        self.ds = nc.Dataset(self.out_file, "w")
+        # defining global attributes
+        self.ds.description = "Results for simulation run on" \
+        f"{t.ctime(t.time())} using PySD version {__version__}"
+        self.ds.model_file = model.py_model_file
+        self.ds.timestep = "{}".format(model.components.time_step())
+        self.ds.initial_time = "{}".format(model.components.initial_time())
+        self.ds.final_time = "{}".format(model.components.final_time())
+
+        # creating variables for all model dimensions
+        for dim_name,coords in model.subscripts.items():
+            coords = np.array(coords)
+            # create dimension
+            self.ds.createDimension(dim_name, len(coords))
+            # length of the longest string in the coords
+            max_str_len = len(max(coords, key=len))
+
+            # create variable
+            # TODO: check if the type could be defined otherwise)
+            var = self.ds.createVariable(dim_name,
+                                    f"S{max_str_len}",
+                                    (dim_name,))
+            # assigning values to variable
+            var[:] = coords
+
+        # creating the time dimension as unlimited
+        self.ds.createDimension("time", None)
+
+        # creating variables in capture_elements
+        self.__create_ds_vars(model, capture_elements)
+
+    def update(self, model, capture_elements):
+        """
+        Writes values of variables in capture_elements in the netCDF4 Dataset.
 
         Parameters
         ----------
-        df: pandas.DataFrame
-            Dataframe to add elements.
+        model: pysd.Model
+            PySD Model object
+        capture_elements: set
+            Which model elements to capture - uses pysafe names.
+
+        Returns
+        -------
+        None
+        """
+        for key in capture_elements:
+            comp = getattr(model.components, key)
+            comp_vals = comp()
+            if isinstance(comp_vals, xr.DataArray):
+                self.ds[key][self.step, :] = comp_vals.values
+            elif isinstance(comp_vals, np.ndarray):
+                self.ds[key][self.step, :] = comp_vals
+            else:
+                self.ds[key][self.step] = comp_vals
+
+        self.step += 1
+
+    def postprocess(self, **kwargs):
+        """
+        Closes netCDF4 Dataset.
+        """
+        self.ds.close()
+
+        if kwargs.get("flatten"):
+            warnings.warn("DataArrays stored in netCDF4 will not be flattened")
 
+        print(f"Results stored in {self.out_file}")
+
+    def add_run_elements(self, model, capture_elements):
```

Comment:
I would add these elements to the netCDF as constants, without including the time dimension in them

Reply:
Thanks for the comments @enekomartinmartinez.

I thought about doing it as you mention, but then I thought that I would reproduce what it's done when the output is a pandas DataFrame. I understand that with DataFrames you don't have the choice, though.

So yeah, I'll make them constants, as you suggest.